buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -588,7 +588,7 @@ static ByteBuf encodeString0(ByteBufAllocator alloc, boolean enforceHeap, CharBu
             dst = alloc.buffer(length);
         }
         try {
-            final ByteBuffer dstBuf = dst.internalNioBuffer(0, length);
+            final ByteBuffer dstBuf = dst.internalNioBuffer(dst.readerIndex(), length);
```

Comment:
thats fine that said it should not make any difference as we just allocated the buffer so the readerIndex() must be 0